collect_app/src/androidTest/java/org/odk/collect/android/formentry/AddRepeatTest.java

```diff
@@ -0,0 +1,97 @@
+package org.odk.collect.android.formentry;
+
+import android.Manifest;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.rule.GrantPermissionRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.activities.MainMenuActivity;
+import org.odk.collect.android.support.CopyFormRule;
+import org.odk.collect.android.support.ResetStateRule;
+import org.odk.collect.android.support.pages.EndOfFormPage;
+import org.odk.collect.android.support.pages.FormEntryPage;
+import org.odk.collect.android.support.pages.MainMenuPage;
+
+@RunWith(AndroidJUnit4.class)
+public class AddRepeatTest {
+
+    private static final String FORM = "one-question-repeat.xml";
```

Comment:
Migth see if I can get that into the `FormController` tests instead as I'd like to avoid setting up anything other than the simplest cases in the "feature tests"

Reply:
Sorry I completely misinterpreted this when I read it on Friday. Basically you're suggesting a test for checking the options menu is setup correctly (and whether `invalidateOptionsMenu` is getting called in the right places). I'll review the coverage and maybe do some mutation testing to see if we're got some blind spots.

Reply:
Yeah after playing around with it feels like we're pretty well covered. I've realised looking at it that we had extra calls to `invalidateOptionsMenu` as we really should just call it every time we call `showView`. 

Also looking at this made me realize it would be nice to pull out some a view (holder) that can encapsulate `createView` and `showView` as it would remove a lot of logic from `FormEntryAcitivty` and would let us test things like "every time we should a view we do X". I'll add this to the list of follow ups that I'll make issues for.